tests/network/localnet/conftest.py

```diff
@@ -83,12 +83,49 @@ def cudn_localnet(
         yield cudn
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
+def cudn_localnet_no_vlan(
+    namespace_localnet_1: Namespace,
+) -> Generator[libcudn.ClusterUserDefinedNetwork]:
+    with localnet_cudn(
+        name=LOCALNET_BR_EX_NETWORK,
+        match_labels=LOCALNET_TEST_LABEL,
+        physical_network_name=LOCALNET_BR_EX_NETWORK,
+    ) as cudn:
+        cudn.wait_for_status_success()
+        yield cudn
+
+
+@pytest.fixture(scope="class")
+def vm_localnet_no_vlan_ip_address(ipv4_localnet_address_pool: Generator[str]) -> str:
+    return next(ipv4_localnet_address_pool)
+
+
+@pytest.fixture(scope="class")
+def vm_localnet_no_vlan(
+    namespace_localnet_1: Namespace,
+    vm_localnet_no_vlan_ip_address: str,
+    cudn_localnet_no_vlan: libcudn.ClusterUserDefinedNetwork,
+) -> Generator[BaseVirtualMachine]:
+    with localnet_vm(
+        namespace=namespace_localnet_1.name,
+        name="localnet-no-vlan-vm",
+        physical_network_name=cudn_localnet_no_vlan.name,
+        spec_logical_network=LOCALNET_BR_EX_NETWORK,
+        cidr=vm_localnet_no_vlan_ip_address,
+    ) as vm:
+        vm.start()
+        vm.wait_for_ready_status(status=True)
+        vm.wait_for_agent_connected()
```

Comment:
these checks appear multiple times in sig-network, you should centarlize under one function

Reply:
I am unsure if there is one correct thing to do in this regard.
To run a VM one needs to 1) define it, 2) start it, 3) check the VM actually started at the domain level and finally 4) that the OS is up and running.
There are pros and cons to separate these steps and not combine them.

Reply:
Can I resolve this thread?